Looks like this is causing trouble:
```
3:54:06 .................................PHP Fatal error:  Uncaught Exception: files_trashbin can't be enabled since it is not installed. in /var/lib/jenkins/workspace/owncloud-core_core_PR-27307-SKLOEDKDM4HKNS6LBRDSJXPYARM2DS2FUT5DNTDNVN4JHC6AFGAA/lib/private/App/AppManager.php:216
13:54:06 Stack trace:
13:54:06 #0 /var/lib/jenkins/workspace/owncloud-core_core_PR-27307-SKLOEDKDM4HKNS6LBRDSJXPYARM2DS2FUT5DNTDNVN4JHC6AFGAA/lib/private/legacy/app.php(366): OC\App\AppManager->enableApp('files_trashbin')
13:54:06 #1 /var/lib/jenkins/workspace/owncloud-core_core_PR-27307-SKLOEDKDM4HKNS6LBRDSJXPYARM2DS2FUT5DNTDNVN4JHC6AFGAA/tests/lib/StreamWrappersTest.php(41): OC_App::enable('files_trashbin')
13:54:06 #2 /var/lib/jenkins/workspace/owncloud-core_core_PR-27307-SKLOEDKDM4HKNS6LBRDSJXPYARM2DS2FUT5DNTDNVN4JHC6AFGAA/lib/composer/phpunit/phpunit/src/Framework/TestSuite.php(758): Test\StreamWrappersTest::tearDownAfterClass()
13:54:06 #3 /var/lib/jenkins/workspace/owncloud-core_core_PR-27307-SKLOEDKDM4HKNS6LBRDSJXPYARM2DS2FUT5DNTDNVN4JHC6AFGAA/lib/composer/phpunit/phpunit/src/Framework/TestSuite.php(753): PHPUnit_Framework_T in /var/lib/jenkins/workspace/owncloud-core_core_PR-27307-SKLOEDKDM4HKNS6LBRDSJXPYARM2DS2FUT5DNTDNVN4JHC6AFGAA/lib/private/App/AppManager.php on line 216
13:54:06 Makefile:177: recipe for target 'test-php' failed
13:54:06 make: *** [test-php] Error 255
```

--- a/lib/private/App/AppManager.php
+++ b/lib/private/App/AppManager.php
@@ -212,6 +212,9 @@ class AppManager implements IAppManager {
 	 * @param string $appId
 	 */
 	public function enableApp($appId) {
+		if(!Installer::isInstalled($appId)) {
+			throw new \Exception("$appId can't be enabled since it is not installed.");
+		}
 		$this->installedAppsCache[$appId] = 'yes';
 		$this->appConfig->setValue($appId, 'enabled', 'yes');
 		$this->dispatcher->dispatch(ManagerEvent::EVENT_APP_ENABLE, new ManagerEvent(

--- a/lib/private/App/AppManager.php
+++ b/lib/private/App/AppManager.php
@@ -30,6 +30,7 @@
 
 namespace OC\App;
 
+use OC_App;
 use OC\Installer;
 use OCP\App\IAppManager;
 use OCP\App\ManagerEvent;
@@ -212,7 +213,7 @@ class AppManager implements IAppManager {
 	 * @param string $appId
 	 */
 	public function enableApp($appId) {
-		if(!Installer::isInstalled($appId)) {
+		if(OC_App::getAppPath($appId) === false) {
 			throw new \Exception("$appId can't be enabled since it is not installed.");
 		}
 		$this->installedAppsCache[$appId] = 'yes';

--- a/lib/private/Installer.php
+++ b/lib/private/Installer.php
@@ -171,7 +171,7 @@ class Installer {
 	 *
 	 * Checks whether or not an app is installed, i.e. registered in apps table.
 	 */
-	public static function 	isInstalled( $app ) {
+	public static function isInstalled( $app ) {
 		return (\OC::$server->getConfig()->getAppValue($app, "installed_version", null) !== null);
 	}
 

--- a/tests/lib/App/ManagerTest.php
+++ b/tests/lib/App/ManagerTest.php
@@ -106,14 +106,27 @@ class ManagerTest extends TestCase {
 
 	public function testEnableApp() {
 		$this->expectClearCache();
-		$this->manager->enableApp('test');
-		$this->assertEquals('yes', $this->appConfig->getValue('test', 'enabled', 'no'));
+		// making sure "files_trashbin" is disabled
+		if ($this->manager->isEnabledForUser('files_trashbin')) {
+			$this->manager->disableApp('files_trashbin');
+		}
+		$this->manager->enableApp('files_trashbin');
+		$this->assertEquals('yes', $this->appConfig->getValue('files_trashbin', 'enabled', 'no'));
 	}
 
 	public function testDisableApp() {
 		$this->expectClearCache();
-		$this->manager->disableApp('test');
-		$this->assertEquals('no', $this->appConfig->getValue('test', 'enabled', 'no'));
+		$this->manager->disableApp('files_trashbin');
+		$this->assertEquals('no', $this->appConfig->getValue('files_trashbin', 'enabled', 'no'));
+	}
+	/**
+	 * @expectedException \Exception
+	 */
+	public function testNotEnableIfNotInstalled() {
+		$this->manager->enableApp('some_random_name_which_i_hope_is_not_an_app');
+		$this->assertEquals('no', $this->appConfig->getValue(
+			'some_random_name_which_i_hope_is_not_an_app', 'enabled', 'no'
+		));
 	}
 
 	public function testEnableAppForGroups() {

